### ISSUE TITLE
Invisible characters screw up 'meteor build' with 'Error: Unexpected character' at future.js:245

### ISSUE THREAD

**Issue by freiit** [meteor/meteor#4586]
Step 1:
$ echo -n $'\x61\x3d\xe2\x80\xa8\x31\x0a' > binary.js # contains now 'a=1' + 3 invisible bytes

Step 2:
Test with node:
$ node binary.js # works fine, no output, no error

Step 3:
$ meteor build /tmp/

    /Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/fibers/future.js:245
                                                    throw(ex);
                                                          ^
    Error: Unexpected character ' ' (line: 1, col: 14, pos: 14)
        at Object.Future.wait (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/fibers/future.js:395:16)
        at /Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/buildmessage.js:556:31
        at Array.forEach (native)
        at Function._.each._.forEach (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:79:11)
        at /Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/buildmessage.js:554:9
        at /Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/buildmessage.js:352:18
        at [object Object]._.extend.withValue (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/fiber-helpers.js:115:14)
        at /Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/buildmessage.js:345:34
        at [object Object]._.extend.withValue (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/fiber-helpers.js:115:14)
        at /Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/buildmessage.js:343:23
        at [object Object]._.extend.withValue (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/fiber-helpers.js:115:14)
        at enterJob (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/buildmessage.js:317:26)
        at Object.forkJoin (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/buildmessage.js:531:3)
        at [object Object]._.extend.minifyJs (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/bundler.js:818:28)
        at [object Object]._.extend.make (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/bundler.js:509:12)
        at /Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/bundler.js:2054:14
        at /Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/bundler.js:2099:20
        at Array.forEach (native)
        at Function._.each._.forEach (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:79:11)
        at /Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/bundler.js:2098:7
        at /Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/buildmessage.js:264:13
        at [object Object]._.extend.withValue (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/fiber-helpers.js:115:14)
        at /Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/buildmessage.js:257:29
        at [object Object]._.extend.withValue (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/fiber-helpers.js:115:14)
        at /Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/buildmessage.js:255:18
        at [object Object]._.extend.withValue (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/fiber-helpers.js:115:14)
        at /Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/buildmessage.js:246:23
        at [object Object]._.extend.withValue (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/fiber-helpers.js:115:14)
        at Object.capture (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/buildmessage.js:245:19)
        at Object.exports.bundle (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/bundler.js:2040:31)
        at buildCommand (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/commands.js:894:30)
        at Command.main.registerCommand._.extend.name [as func] (/Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/commands.js:746:12)
        at /Users/dummy/.meteor/packages/meteor-tool/.1.1.3.1sb878t++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/tools/main.js:1363:23

**Comment by glasser** [comment]
OK, this character is [LINE SEPARATOR](http://www.fileformat.info/info/unicode/char/2028/index.htm) which is supposed to be legal in JS.  Looks like UglifyJS (the minifier) is choking on it.  In fact if we just run uglifyjs directly we get the same error:

```
$ ~/Projects/Meteor/meteor/packages/minifiers/.npm/package/node_modules/.bin/uglifyjs  binary.js  
Parse error at binary.js:1,2
Unexpected character ' '
Error
    at new JS_Parse_Error (/Users/glasser/Projects/Meteor/meteor/packages/minifiers/.npm/package/node_modules/uglify-js/lib/parse.js:196:18)
    at js_error (/Users/glasser/Projects/Meteor/meteor/packages/minifiers/.npm/package/node_modules/uglify-js/lib/parse.js:204:11)
    at parse_error (/Users/glasser/Projects/Meteor/meteor/packages/minifiers/.npm/package/node_modules/uglify-js/lib/parse.js:313:9)
    at Object.next_token [as input] (/Users/glasser/Projects/Meteor/meteor/packages/minifiers/.npm/package/node_modules/uglify-js/lib/parse.js:561:9)
    at next (/Users/glasser/Projects/Meteor/meteor/packages/minifiers/.npm/package/node_modules/uglify-js/lib/parse.js:660:25)
    at maybe_assign (/Users/glasser/Projects/Meteor/meteor/packages/minifiers/.npm/package/node_modules/uglify-js/lib/parse.js:1424:17)
    at expression (/Users/glasser/Projects/Meteor/meteor/packages/minifiers/.npm/package/node_modules/uglify-js/lib/parse.js:1440:20)
    at simple_statement (/Users/glasser/Projects/Meteor/meteor/packages/minifiers/.npm/package/node_modules/uglify-js/lib/parse.js:888:55)
    at /Users/glasser/Projects/Meteor/meteor/packages/minifiers/.npm/package/node_modules/uglify-js/lib/parse.js:761:19
    at /Users/glasser/Projects/Meteor/meteor/packages/minifiers/.npm/package/node_modules/uglify-js/lib/parse.js:721:24
```

However this is already fixed on devel by upgrading UglifyJS from 2.4.17 to 2.4.20.

**Comment by freiit** [comment]
Thanks for sorting this out! :)
